### DISCUSSION TITLE
Help configuring Google Chrome and Outlook

### DISCUSSION THREAD

**Discussion by MonkeyMysteryCoding** [cmu-sei/GHOSTS#225] [Q&A]
I am completely new to this type of stuff and am trying to further my education, however, I am just completely lost on setting all this stuff up. Is there any way someone can run me through setting up Google Chrome or Outlook or show me some examples. Is Google Chrome set up the same way as Firefox or slightly different? I look at the documentation, but it just still does not work. Thanks

**Comment by sei-dupdyke** [comment]
Hi, thanks for checking us out. The default client install should _just work_. When you say it does not, are you seeing an error? Are there any entries in the application.log file?

**Reply by MonkeyMysteryCoding** [reply]
Wow, I did not know there were any log entries. I will check that out right now. It should send emails by default? Do I need to set anything up in Outlook. Also, it does open up google chrome, but it is stuck on the about:blank thing. 
I made the handler this
"HandlerType": "BrowserChrome".
"Initial": "http://www.google.com",
"UtcTimeOn": "00:00:00",
"UtcTimeOff": "00:00:00",
"Loop": True,
"TimeLineEvents": [
{
"Command": "random",
"CommandArgs": [
"http://reddit.com",
"http://instagram.com",
"http://facebook.com",
],
"DelayAfter" : 1000,
"DelayBefore": 0
}
]
}

**Comment by sei-dupdyke** [comment]
Oh, that is likely the wrong chromedriver for the version of chrome you are running. You will have to get the appropriate version for your version of Chrome from here https://chromedriver.chromium.org/downloads and replace chromedriver.exe in the ghosts executable folder. Then restart ghosts and it should start browsing.

**Reply by MonkeyMysteryCoding** [reply]
Got it, Thanks! Also, how is OutLook supposed to automatically work if there is no Handler?

**Reply by sei-dupdyke** [reply]
Outlook takes a bit of configuration to work, which is why we don't enable it by default. You have to set it up to be able to send/receive from the mailbox you want to use. Once that is good to go, you have to install the Redemption component included with GHOSTS via rgsvr command.

**Reply by MonkeyMysteryCoding** [reply]
Do you know where I can find more instructions on that? So far I created an Active Directory with a mail server (Mail Enable). Do I just change the probability values and add the emails to the list and login to OutLook?

**Reply by MonkeyMysteryCoding** [reply]
Also, my ChromeDriver is 114.0.5735.90 and Chrome is 114.0.5735.248. Isn't that the same version?

**Reply by sei-dupdyke** [reply]
What does ./logs/application.log say?

**Reply by MonkeyMysteryCoding** [reply]
I got the Chrome working, but I just don't know where to start with the OutLook stuff.

**Reply by sei-dupdyke** [reply]
Does Outlook work without ghosts (e.g. can you just go and use it directly, without ghosts running)?

**Reply by MonkeyMysteryCoding** [reply]
Kind of, I am creating my own mail server, even if it doesn't synch up to the web mail client it still sends something. Basically, the mail sends, but it doesn't actually reach the other user, so I feel like it should still work. Also, is there any possible way to push out changes in the configuration from the API server to all the NPCs at once as in manage it centrally.

**Reply by sei-dupdyke** [reply]
Not sure on the email thing. Managing clients from the API is covered here: https://cmu-sei.github.io/GHOSTS/core/api/timelines/

**Reply by MonkeyMysteryCoding** [reply]
I think I'm going to take a break from the Outlook stuff because I also need to setup my own email server which is making it 10x harder. However, for the command handler, if I wanted to use NMAP, would it be "Command" : "\CD:\\Program Files (x86)\\Nmap\\nmap.exe\" -V 172.18.18.5/24" as the right format? Because it doesn't seem to be working
